src/uu/hashsum/src/hashsum.rs

```diff
@@ -344,38 +221,92 @@ pub fn uumain(mut args: impl uucore::Args) -> UResult<()> {
         .unwrap_or(&false);
     let status = matches.get_flag("status");
     let quiet = matches.get_flag("quiet") || status;
-    let strict = matches.get_flag("strict");
+    //let strict = matches.get_flag("strict");
     let warn = matches.get_flag("warn") && !status;
-    let zero = matches.get_flag("zero");
+    let zero: bool = matches.get_flag("zero");
     let ignore_missing = matches.get_flag("ignore-missing");
 
     if ignore_missing && !check {
         // --ignore-missing needs -c
-        return Err(HashsumError::IgnoreNotCheck.into());
+        return Err(ChecksumError::IgnoreNotCheck.into());
     }
 
     let opts = Options {
-        algoname,
-        digest: algo,
-        output_bits: bits,
+        algoname: algo.name,
+        digest: (algo.create_fn)(),
+        output_bits: algo.bits,
         binary,
-        check,
+        //check,
         tag,
         nonames,
-        status,
-        quiet,
-        strict,
-        warn,
+        //status,
+        //quiet,
+        //strict,
+        //warn,
         zero,
-        ignore_missing,
+        //ignore_missing,
     };
 
-    match matches.get_many::<OsString>("FILE") {
+    if check {
+        let text_flag: bool = matches.get_flag("text");
+        let binary_flag: bool = matches.get_flag("binary");
+        let strict = matches.get_flag("strict");
+
+        if (binary_flag || text_flag) && check {
```

Comment:
`check` is always `true` at this point, thus it is unnecessary.
```suggestion
        if binary_flag || text_flag {
```